source/common/upstream/logical_dns_cluster.cc

```diff
@@ -35,12 +35,8 @@ LogicalDnsCluster::LogicalDnsCluster(
                            : Config::Utility::translateClusterHosts(cluster.hosts())) {
   const auto& locality_lb_endpoints = load_assignment_.endpoints();
   if (locality_lb_endpoints.size() != 1 || locality_lb_endpoints[0].lb_endpoints().size() != 1) {
-    if (cluster.has_load_assignment()) {
-      throw EnvoyException(
-          "LOGICAL_DNS clusters must have a single locality_lb_endpoint and a single lb_endpoint");
-    } else {
-      throw EnvoyException("LOGICAL_DNS clusters must have a single host");
```

Comment:
nit: Since we keep the `hosts` field, can we keep this and convert the test to v2 as well? A relevant comment from Alyssa: https://github.com/envoyproxy/envoy/pull/3261#discussion_r188281502

Reply:
This is already covered here: https://github.com/envoyproxy/envoy/blob/master/test/common/upstream/logical_dns_cluster_test.cc#L293

Reply:
Hum, yes sorry, but it will be removed by this PR? https://github.com/envoyproxy/envoy/pull/7681/files#diff-331525e48cab6701d9dbf53871db55a4L304. Sorry if it wasn't clear, as per [Alyssa's comment](https://github.com/envoyproxy/envoy/pull/3261#discussion_r188281502), we want to have a different error for each case (i.e. when the config uses `hosts` and `load_assignment`).

Reply:
Brought back dual error message.

Reply:
Thanks!